File: CS/Form1.cs
using System;
using System.Drawing.Design;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Data.Browsing;
using DevExpress.XtraBars;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Design;

namespace WindowsApplication1 {
    public partial class Form1 : DevExpress.XtraReports.UserDesigner.XRDesignFormEx {
        static BarButtonItem bbiCustomControl;
        static Form1() {
        }
        public Form1() {
            InitializeComponent();
            XtraReport1 rep = new XtraReport1();
            rep.DesignerLoaded += new DevExpress.XtraReports.UserDesigner.DesignerLoadedEventHandler(rep_DesignerLoaded);
            this.xrDesignPanel.OpenReport(rep);

        }
        void rep_DesignerLoaded(object sender, DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs e) {
            e.DesignerHost.RemoveService(typeof(IFieldListDragDropService));
            e.DesignerHost.AddService(typeof(IFieldListDragDropService),new MyFieldListDragDropService(e.DesignerHost));
        }
    }
    public class MyFieldListDragDropService : IFieldListDragDropService{
        IDesignerHost host;
        public MyFieldListDragDropService(IDesignerHost host){this.host = host;}
        public IDragHandler GetDragHandler() {
            return new MyFieldDragHandler(host);
        }
    }
    public class MyFieldDragHandler : FieldDragHandler {
        IDesignerHost host;
        public MyFieldDragHandler(IDesignerHost host)
            : base(host) {
            this.host = host;
        }
        static bool MIsContolContainer(XRControl control) {
            return control is Band || control is XRPanel;
        }
        public override void HandleDragDrop(object sender, DragEventArgs e) {

            Point dropPoint = new Point(e.X, e.Y);
            DataInfo[] droppedData = e.Data.GetData(typeof(DataInfo[])) as DataInfo[];
            XRControl control = bandViewSvc.GetControlByScreenPoint(dropPoint);
            
            if(MIsContolContainer(control)) {
                Size size = new Size(100,25);
                Point location = bandViewSvc.PointToClient(dropPoint, control);
                XRRichText l = null;
                for(int i = 0; i < droppedData.Length; i++) {
                    l = new XRRichText();
                    l.Size = size;
                    l.Location = location;
                    l.DataBindings.Add("Rtf", droppedData[i].Source, droppedData[i].Member);
                    ReportDesigner.AddToContainer(host, l);
                    control.Controls.Add(l);
                    location.X += 100;
                }
            }
            else
                base.HandleDragDrop(sender, e);

        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing.Design
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Data.Browsing
Imports DevExpress.XtraBars
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Design

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits DevExpress.XtraReports.UserDesigner.XRDesignFormEx
		Private Shared bbiCustomControl As BarButtonItem
		Shared Sub New()
		End Sub
		Public Sub New()
			InitializeComponent()
			Dim rep As New XtraReport1()
			AddHandler rep.DesignerLoaded, AddressOf rep_DesignerLoaded
			Me.xrDesignPanel.OpenReport(rep)

		End Sub
		Private Sub rep_DesignerLoaded(ByVal sender As Object, ByVal e As DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs)
			e.DesignerHost.RemoveService(GetType(IFieldListDragDropService))
			e.DesignerHost.AddService(GetType(IFieldListDragDropService),New MyFieldListDragDropService(e.DesignerHost))
		End Sub
	End Class
	Public Class MyFieldListDragDropService
		Implements IFieldListDragDropService
		Private host As IDesignerHost
		Public Sub New(ByVal host As IDesignerHost)
			Me.host = host
		End Sub
		Public Function GetDragHandler() As IDragHandler Implements IFieldListDragDropService.GetDragHandler
			Return New MyFieldDragHandler(host)
		End Function
	End Class
	Public Class MyFieldDragHandler
		Inherits FieldDragHandler
		Private host As IDesignerHost
		Public Sub New(ByVal host As IDesignerHost)
			MyBase.New(host)
			Me.host = host
		End Sub
		Private Shared Function MIsContolContainer(ByVal control As XRControl) As Boolean
			Return TypeOf control Is Band OrElse TypeOf control Is XRPanel
		End Function
		Public Overrides Sub HandleDragDrop(ByVal sender As Object, ByVal e As DragEventArgs)

			Dim dropPoint As New System.Drawing.Point(e.X, e.Y)
			Dim droppedData() As DataInfo = TryCast(e.Data.GetData(GetType(DataInfo())), DataInfo())
			Dim control As XRControl = bandViewSvc.GetControlByScreenPoint(dropPoint)

			If MIsContolContainer(control) Then
				Dim size As New Size(100,25)
				Dim location As System.Drawing.Point = bandViewSvc.PointToClient(dropPoint, control)
				Dim l As XRRichText = Nothing
				For i As Integer = 0 To droppedData.Length - 1
					l = New XRRichText()
					l.Size = size
					l.Location = location
					l.DataBindings.Add("Rtf", droppedData(i).Source, droppedData(i).Member)
					ReportDesigner.AddToContainer(host, l)
					control.Controls.Add(l)
					location.X += 100
				Next i
			Else
				MyBase.HandleDragDrop(sender, e)
			End If

		End Sub
	End Class
End Namespace